utilities/transactions/write_unprepared_txn.cc

```diff
@@ -206,13 +208,13 @@ Status WriteUnpreparedTxn::FlushWriteBatchToDB(bool prepared) {
       db_impl_->immutable_db_options().two_write_queues);
   const bool DISABLE_MEMTABLE = true;
   uint64_t seq_used = kMaxSequenceNumber;
-  // We only set log_number_ if it hasn't been set, so that log_number_ always
-  // refers to the oldest log file referenced by this transaction. This helps
-  // ensure that log files containing batches from this transaction are not
-  // deleted.
+  // log_number_ should refer to the oldest log containing uncommitted data
+  // from the current transaction. This means that if log_number_ is set,
+  // WriteImpl should not overwrite that value, so set log_used to nullptr if
+  // log_number_ is already set.
+  uint64_t* log_used = log_number_ ? nullptr : &log_number_;
```

Comment:
nit: s/log_used/log_number/g

Reply:
I think I'll keep log_used, since that's the name of the parameter to WriteImpl, and it might be hard to spot the difference between "log_number_" and "log_number".